x-pack/plugins/ml/public/application/data_frame_analytics/common/use_results_view_config.ts

```diff
@@ -97,7 +97,7 @@ export const useResultsViewConfig = (jobId: string) => {
 
           try {
             const destIndex = getDestinationIndex(jobConfigUpdate);
-            const destDataViewId = (await getDataViewIdFromName(destIndex)) ?? destIndex;
+            const destDataViewId = (await getDataViewIdFromName(dataViews, destIndex)) ?? destIndex;
```

Comment:
Same note here for either instantiating `indexServiceFactory` with the necessary services available via `useMlKibana` or passing in the service or function after instantiation wherever this hook is used.

Reply:
Updated in [70ab770](https://github.com/elastic/kibana/pull/176788/commits/70ab770cc5e06985dba0dd7b86123eee21f7a72e).